lib/levels.js

```diff
@@ -89,10 +82,6 @@ function setLevel (level) {
   const useOnlyCustomLevelsVal = this[useOnlyCustomLevelsSym]
 
   for (var key in values) {
-    if (levelVal > values[key]) {
-      this[key] = noop
-      continue
-    }
```

Comment:
This change is going to slow us down significantly in case a log level is disabled. Changing this is an hard no for me.

Reply:
are you sure this slowing anything down _significantly_? 
Effectively this will simply replace `() => {}` with 
```
    if (levelValOfMethod < levels[this[rootLevelSym]]) {

      // don't write the log if this method's log severity is lower than what the logger is configured to write
      return;
    }
```

where `levelValOfMethod < levels[this[rootLevelSym]]` is the "expensive" part if you will. This will walk up the prototype chain to get property and then do a numerical comparison. I understand that this adds some extra computation but I doubt this is slowing things down _significantly_ . More than happy to benchmark this though.

Reply:
using a no-op and specifically _the same no-op_ is a deliberate decision to provide a good hint to Turbofan so that it can optimize disabled log statements away.